Cambio de icono y Navegación

JLeonN opened the pull request:
Probando ....
que ponemos aca los programadores por lo general ?

JLeonN committed:
Probando.

Probando GitHub Desktop y como descargar un Repositorios

JLeonN committed:
Cambio de icono

Probando enviar cambios en un proyecto en equipo.

JLeonN committed:
Nav

Instalación de NPM, comencé con el nav, también pude algo de CSS, pero es prueba.

JLeonN committed:
Error

Estoy teniendo un error con las ubicaciones, pero estoy seguro de que eso está bien.

En el archivo main.jsx, agrego una línea y me da el error. Lo comento, pero el error sigue.

francosang commented on frontend/dist/index.html:
```diff
@@ -2,11 +2,12 @@
   <head>
     <meta charset="UTF-8" />
-    <link rel="icon" type="image/svg+xml" href="/icon.svg" />
+    <!-- <link rel="icon" type="image/svg+xml" href="/frontend/public/Cowboys.png" /> -->
```
Por favor, revertir estos cambios en este archivo.

No se deben modificar los archivos en la carpeta `frontend/dist/`.

Los archivos de la carpeta `frontend/dist/` son autogenerados por react, y solo se modifican cuando se quiere publicar una actualizacion de la app.

francosang commented on frontend/index.html:
```diff
@@ -2,9 +2,13 @@
   <head>
     <meta charset="UTF-8" />
-    <link rel="icon" type="image/svg+xml" href="/icon.svg" />
+    <!-- <link rel="icon" type="image/svg+xml" href="/icon.svg" /> -->
```
No dejar lineas comentadas que ya no se necesitan. 

Por favor borrar esta linea y ademas borrar el icono viejo.

francosang replied:
@JLeonN Hay que borrar el icono viejo, ya no se usa.

francosang commented on frontend/index.html:
```diff
@@ -2,9 +2,13 @@
     <title>The Cowboy of the Day</title>
+
+    <!-- Icono de titulo -->
+    <link rel="icon" type="imag/x-icon" href="/frontend/public/Cowboys.png">
```
La ubicacion del icono anterior era`/icon.svg`, el nuevo icono deberia funcionar en la misma ubicacion.

Por favor cambiar a `/Cowboys.png`.

francosang commented on package.json:
```diff
@@ -11,6 +11,7 @@
     "express": "^4.18.2",
-    "pg": "^8.11.3"
+    "pg": "^8.11.3",
+    "react-router-dom": "^5.3.0"
```
Revertir este cambio. 
Este archivo `package.json` es para el servidor y la API.
El `package.json` del frontend esta en `frontend/package.json`.

Esta es la causa del error de los imports, la libreria de `react-router-dom` fue instalada en el servidor en lugar de instalarse en el frontend, por eso falla diciendo que no conoce esa libreria.

Verificar el directorio correcto donde instalar la dependencia:
```shell
cd frontend
npm install react-router-dom
```

francosang commented on server/index.js:
```diff
@@ -1,3 +1,7 @@
+
+
+
+
```
Borrar estas lineas en blanco.

francosang commented on frontend/src/App.jsx:
```diff
@@ -25,20 +27,27 @@ function App() {
-          )
-        }
-      </div>
+      {/* <Router> */}
```
Como vas a ordenar estos componentes?

Creo que deberiamos organizar todo asi:
 - Un elemento fijo al pricipio que sea el Nav, asi aparece siempre arriba.
 - Los Routes y Route abajo, asi siempre aparecen abajo del nav.
 - Un componente por pagina de nuestra app, asi queda todo mejor estructurado.

Algo asi:
```
┌──────────────────────────┐
| < Nav >                  │
| < Rutas >                |
|    < CowboyDelDia />     |
|    < ElMasTonto   />     |
|    < Estadisticas />     |
└──────────────────────────┘
```
Dependiendo del link y la URL iriamos cambiando los componentes.

Que tenias en mente? Que te parece esto?

francosang reviewed (commented):
Un exito @JLeonN, re bien los cambios 🎉 ! 

Fijate los comentarios que te deje, uno esta relacionado con el error que hablamos por discord.

Felicidades por tu primer PR!

JLeonN committed:
Reparando el daño

Reparando lo que Franco me pidió, creo que esta vez quedó todo bien.
Lo que sigo sin poder cambiar es el icono.

JLeonN committed:
Nav

Mejorando el Nav y el CSS, también coloqué más tipografías.

JLeonN committed:
Modificando CSS

He notado que el CSS estaba quedando desordenado. Actualmente, estoy considerando dos opciones: crear un archivo CSS exclusivo para cada componente o tener un único archivo que contenga únicamente los estilos de los componentes. Todavía no he decidido cuál es la mejor opción y estoy evaluando ambas posibilidades.

JLeonN committed:
Contacto

He creado un componente de contacto y actualmente estoy trabajando en él. Además, estoy comenzando a trabajar en la navegación.

JLeonN committed:
Mas componentes

Mas componentes

      <NavBar />
      <Inicio />
      <Acerca />
      <Estadisticas />
      <Contacto />

JLeonN committed:
Navegacion pronto

La navegación está lista. Avísame si el 'nav' quedó bien fuera de la navegación. Si todo está OK, sigo con lo siguiente.

JLeonN committed:
Error pequeño

Un pequeño error inofensivo. Eso me pasa por no revisar antes de mandar, jaja...

JLeonN committed:
Estilos para los botones

Estoy añadiendo estilos a los botones y estoy utilizando los mismos que tú colocaste.
¡Realmente lucen geniales!

JLeonN committed:
Cambiando el Nav de ubicación.

Estoy haciendo algunos cambios en el Nav. Decidí cambiar su ubicación porque planeo agregar más archivos relacionados a él. Ahora lo mantendré todo organizado en una carpeta exclusiva para todo lo relacionado con el NavBar.

JLeonN committed:
Error de ubicación

😋

francosang commented on frontend/dist/Cowboys.png:
Ahora no te preocupes mucho por esto, pero para proximas modificaciones acordate que carpeta `frontend/dist/` es autogenerada.

No es necesario modificar lo que esta en `frontend/dist/` porque luego lo autogeneramos con un comando de react.

francosang commented on frontend/dist/index.html:
Acordate que no tienen que haber modificaciones en la carpeta `frontend/dist/`.
La carpeta `frontend/dist/` es autogenerada.

francosang commented on frontend/src/App.jsx:
```diff
@@ -25,6 +27,8 @@ function App() {
 
   return (
     <>
+    <NavBar />
```
Corregir indentancion, falta una tabulacion hacia la derecha.
```suggestion
      <NavBar />
```

francosang replied:
Te recomiendo que te instales un extension de auto correccion de codigo para Visual Code.
Estas extensiones le dan formato automatico al codigo, se hace solo.

Yo tengo esta:
 - https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode

Un videito: https://www.youtube.com/watch?v=OmwrOgUXWBs&ab_channel=divcode

francosang commented on frontend/src/Componentes/Estructura.jsx:
```diff
@@ -0,0 +1,21 @@
+import React from "react";
```
Este import no se usa.

francosang commented on package.json:
Revertir estos cambios.

francosang commented on package-lock.json:
Revertir los cambios en este archivo.

francosang reviewed (commented):
Vamoooo 💪 !

Me gustó lo que hiciste con `NavBar` y `Estructura`, y los componentes de la carpeta pagina. 

Re bien 👌 !

Gracias por colaborar en el projecto ❤️ !

francosang reviewed (approved):
🚀

JLeonN committed:
Formato al documento

Estoy dando formato al documento. Siguiendo tu recomendación, instalé el PRETTIER, ¡y el resultado fue impresionante! Quedó todo impecable, mucho mejor de lo que esperaba. Realmente, el PRETTIER es una herramienta excelente.

JLeonN committed:
Cambio de icono exitoso

Por fin logré lo primero que me propuse cambiar el icono. 😁
